packages/terra-table/src/Table.jsx

```diff
@@ -0,0 +1,291 @@
+import React, {
+  useState, useContext, useRef, useCallback, useEffect, useMemo,
+} from 'react';
+import PropTypes from 'prop-types';
+import { injectIntl } from 'react-intl';
+import classNames from 'classnames/bind';
+import ResizeObserver from 'resize-observer-polyfill';
+import ThemeContext from 'terra-theme-context';
+import VisuallyHiddenText from 'terra-visually-hidden-text';
+import ColumnHeader from './subcomponents/ColumnHeader';
+import Row from './subcomponents/Row';
+import rowShape from './proptypes/rowShape';
+import { columnShape } from './proptypes/columnShape';
+import ColumnContext from './utils/ColumnContext';
+import validateRowHeaderIndex from './proptypes/validators';
+import styles from './Table.module.scss';
+
+const cx = classNames.bind(styles);
+
+const propTypes = {
+  /**
+     * String that identifies the element (or elements) that labels the table.
+     */
+  ariaLabelledBy: PropTypes.string,
+
+  /**
+     * String that labels the table for accessibility. If ariaLabelledBy is specified, ariaLabel will not be used.
+     */
+  ariaLabel: PropTypes.string,
+
+  /**
+     * String that will be used to identify the table. If multiple tables are on the same page, each table should have
+     * a unique id.
+     */
+  id: PropTypes.string.isRequired,
+
+  /**
+     * Data for content in the body of the table. Rows will be rendered in the order given.
+     */
+  rows: PropTypes.arrayOf(rowShape),
+
+  /**
+     * Data for pinned columns. Pinned columns are the stickied leftmost columns of the table.
+     * Columns will be presented in the order given.
+     */
+  pinnedColumns: PropTypes.arrayOf(columnShape),
+
+  /**
+     * Data for overflow columns. Overflow columns are rendered in the table's horizontal overflow.
+     * Columns will be presented in the order given.
+     */
+  overflowColumns: PropTypes.arrayOf(columnShape),
```

Comment:
This does not have to be addressed in this JIRA, but I wonder if we should update our pattern to just have a single columns array and have consumers pass the pinned column count as a prop.  We really don't need two separate arrays and end up needing to combine them.  I understand for this JIRA that it is existing code.